py/mpstate.h

```diff
@@ -102,6 +102,15 @@ typedef struct _mp_state_mem_t {
     // This is a global mutex used to make the GC thread-safe.
     mp_thread_mutex_t gc_mutex;
     #endif
+
+    #if MICROPY_GC_PERMANENT_ALLOCS
+    // List of pointers to "always marked" allocs.
+    void **gc_permanent;
```

Comment:
Currently this implementation looks suboptimal.  I think there are 2 options to improve it:
1. change this `void *` to `MICROPY_GC_STACK_ENTRY_TYPE` and store block numbers directly rather than pointers (so on most targets that will halve the size of these entries)
2. keep them as `void *` entries and then move this pointer into the root pointer section of the mp_state_t struct, so that it will automatically be scanned, and scanned recursively, so the additional code in `gc_sweep` is much simpler)

Reply:
OK, I'll go with (1) as we explicitly wanted to make these not scanned recursively.

Reply:
Done.